blocks/fragment/fragment.js

```diff
@@ -13,6 +13,12 @@ export default async function decorate(block) {
     const fragmentSection = fragment.querySelector(':scope .section');
     if (fragmentSection) {
       block.closest('.section').classList.add(...fragmentSection.classList);
+      fragmentSection.childNodes.forEach((childNode) => {
```

Comment:
I am curious as to why is this `section-container` nesting happening at all. I have a hunch that  `decorateSections` is getting called twice somehow, once for the main page and then again for the fragment. I think we should address this in the [decorateSections](https://github.com/hlxsites/sunstar/blob/8f4703c6dd328e9ea8f6eabb8a222d0614c8205e/scripts/lib-franklin.js#L322) to avoid injecting `section-container` if there's already one in the hierarchy. wdyt?

Reply:
Problem is not with the decorateSections but with the [line](https://github.com/hlxsites/sunstar/blob/main/blocks/fragment/fragment.js#L16) where kind of merging is going on for the two sections.
I have included a check here. I am not sure if the else condition is triggered in any scenario, but I added it for safety.